### PR TITLE
Removed temporary partial movie file in scenefilewriter

### LINKED ISSUE #486: Removing temp ffmpeg's temp-files
Before partial movie files are generated by ffmpeg, scenefilewriter creates temp partial movie file and then move them and rename them with their definitive name (which is the hash of the animation). 
Why ? Why not rendering directly partial movie files? 
I think we should, if there is no objection, remove this and make ffmpeg directly render partial movie files.

### PR DESCRIPTION
**Pull request by farahash**
## List of Changes

- Removed temp partial movie file from SceneFileWriter (ffmpeg directly renders partial movie files instead of temp files)

## Motivation

Issue #486 

Closes #486 

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [X] I have added an entry describing the changes from this PR to the [Changelog](https://docs.manim.community/en/latest/changelog.html) at `docs/source/changelog.rst`

### CODE REVIEW

**Review by leotrs — COMMENTED** [review]
Hi @farahash ! Thank you for your contribution. This all looks good to me in general. There is only one thing I want to ask. In this line [right here](https://github.com/ManimCommunity/manim/pull/817/files#diff-1a4d3d0a888b8883d39da0097f2ce872390698a75267d68ab073aa32f8258425L346), the code makes sure to force the file extension by using `config["movie_file_extension"]`. However, the current PR does not force this extension (it just uses `file_path` directly).

This `file_path` is gotten from `self.partial_movie_files` which is a list populated elsewhere. Are we super duper sure that `self.partial_movie_files` contains file names with the correct extension? I believe @huguesdevimeux may know the answer.

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -207,3 +207,4 @@ Other Changes
 #. Rename package from manimlib to manim
 #. Move all imports to :code:`__init__`, so :code:`from manim import *` replaces :code:`from manimlib.imports import *`
 #. Global dir variable handling has been removed. Instead :code:`initialize_directories`, if needed, overrides the values from the cfg files at runtime.
+#. Removed temp partial movie file from SceneFileWriter (ffmpeg directly renders partial movie files instead of temp files)
```
[the remark this answers is not in the document]
Comment (farahash) [reply docs/source/changelog.rst]: Fixed

**Comment by farahash** [comment]
> Hi @farahash ! Thank you for your contribution. This all looks good to me in general. There is only one thing I want to ask. In this line [right here](https://github.com/ManimCommunity/manim/pull/817/files#diff-1a4d3d0a888b8883d39da0097f2ce872390698a75267d68ab073aa32f8258425L346), the code makes sure to force the file extension by using `config["movie_file_extension"]`. However, the current PR does not force this extension (it just uses `file_path` directly).
> 
> This `file_path` is gotten from `self.partial_movie_files` which is a list populated elsewhere. Are we super duper sure that `self.partial_movie_files` contains file names with the correct extension? Or should we be forcing the extension just to make sure? I believe @huguesdevimeux may know the answer.

It seems like add_partial_movie_file() sets the extension using config["movie_file_extension"] so I think we're okay but @huguesdevimeux  can confirm that

**Comment by huguesdevimeux** [comment]
I confirm, `add_partial_movie_file` already take care of the extension format :)

**Review by huguesdevimeux — APPROVED** [review]
SGTM

**Review by leotrs — APPROVED** [review]

**Pull request merged**